### ISSUE TITLE
Suggestions 

### ISSUE THREAD

**Issue by Mahmood1968** [dell-ai-engineering/BigDL-ImageProcessing-Examples#4]
1. **shuffle Datasets randomly:**  Since we are using mini batches with specific batch size, it is important to shuffle for each epoch to have different data for each batch.  Look at this paper : https://arxiv.org/pdf/1709.10432.pdf 
2. **add batch normalization function in python** : here good post which explain whey we need batch normalization : https://towardsdatascience.com/batch-normalization-in-neural-networks-1ac91516821c
3. **up /down sampling** : in order to remove imbalanced problem there is two ways to do that either up sampling the classes with less number of samples or down sample the classes with more number of samples. That means , make the number of samples in all classes are the same . look at this post: https://stats.stackexchange.com/questions/122409/why-downsample
and this 
http://www.simafore.com/blog/handling-unbalanced-data-machine-learning-models

**Comment by balacs13** [comment]
From Yuhao: 1) No need to shuffle the dataset. 2) ResNet does the batch normalization.

**Comment by balacs13** [comment]
3) No longer required to be fixed for this project.
